DatadogSessionReplay/Sources/DatadogSessionReplay/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIImageViewRecorder.swift

```diff
@@ -76,22 +76,28 @@ class ImageDataProvider {
         case .none:
             base64s[hash] = .loading
 
-            DispatchQueue.main.async { [weak self] in
+            DispatchQueue.global(qos: .background).async { [weak self] in
                 print("🏞️💾Size of Memory Cache \(Double(MemoryLayout.size(ofValue: self?.base64s))/1024) KB")
                 let data: Data?
-                data = imageView.asImage().jpegData(compressionQuality: 0.5)
-//                if let tintColor = tintColor, #available(iOS 13.0, *) {
-//                    data = image.withTintColor(tintColor).pngData()
-//                } else {
-//                    data = image.pngData()
-//                }
+                if let tintColor = tintColor, #available(iOS 13.0, *) {
+                    data = image.withTintColor(tintColor).pngData()
+                } else {
+                    data = image.pngData()
+                }
                 let sizeKB = Double(data?.count ?? 0) / 1024.0
                 if let base64String = data?.base64EncodedString(), sizeKB < 128.0 { // Max image size of 128KB
```

Comment:
you don't have to check the size first this way you do not have to compute the base64 for nothing ?